### DISCUSSION TITLE
Create an "auto active on launch" option...

### DISCUSSION THREAD

**Discussion by ChadThomsen** [tom-englert/RegionToShare#31] [Ideas]
I use this tool quite often and have saved it to my windows task bar.  So to launch it and get it to work I have to do the following.

1. Click on the Task Bar Shortcut  to get it to launch, and it snaps to the last place used on the screen which is what I want. 
2. Click on the Task Bar Icon again to make its o that Play Button appears.  (Step one does not reveal the play button, which is odd to me)
3. Click on the Play button. 

What I want to happen.
1.  Click on the Task Bar Shortcut.  It launches, snaps to last location remembered, and is ready to share in teams (or other meeting app such as WebEx).

Not that big of a deal, but when I am taking notes, talking on a TEAMs meeting to a bunch of folks, and trying to get this started all at the same time having to do multiple steps just makes it a bit clunky to start.   This would be a "nice to have" feature.   Keep up the great work as I really like this software!!  Thanks!

**Comment by tom-englert** [comment]
I can't reproduce 2. - when I click the Task Bar shortcut, it just shows up and I can share & click the start button.

**Reply by ChadThomsen** [reply]
Item 2. has seemed to resolve itself.  Maybe something unique to my laptop.    Still would like to have the option to do this:

Click on the Task Bar Shortcut. It launches, snaps to last location remembered, and is ready to share in teams (or other meeting app such as WebEx).

Maybe make a toggle option that you can control in the config file? 
Thanks for your hard work as I really like this utility.

**Reply by tom-englert** [reply]
So you really want to trade in this one click for scrolling and searching in Teams?

**Comment by ChadThomsen** [comment]
I am not sure what you mean by that but I will explain my process a bit better for clarity.  First off I have a short cut for  "Region to Share" and its saved to my task bar. 
1. Start/join meeting in teams.
2. Click on the Task Bar Shortcut to start "Region to Share"
3. Click Play on "Region to Share"  (Window size predefined as it snaps to my FancyZones windows via Windows Power Toys)
4. Go into teams  and Click share, then click "Window" in the pop up, and last click "Region To Share".
5. Now sharing my region to meeting members

My request is to to eliminate step #3.  In this case you would start "Region to Share" and it would automagically start sharing the Window with no play button.   It would snap to the last window size/location used or if using something like FancyZones, it would snap to the last FancyZone used.  

What would be ideal would be to just have it so I could set "Region to Share" as a default option in Teams.  That way I click the share  screen button in teams it would launch the "Region to Share" window, and the window size would be what ever it was the last time it was in use.  I realize this is most likely not possible since that would mean altering the code in Teams itself.  Hopefully MS will incorporate your nice code into Teams some day assuming that is allowed. 

Keep up the great work as I really like and utilize this utility on a daily basis as I work from home and do several Teams meetings a day that involve screen sharing.  With an ultrawide monitor your utility makes it much better.

**Reply by tom-englert** [reply]
When you first do step 4 and then 3, RTS is the first window in the teams list, and clearly distinguishable by the green hatch. 

If you do it 3 then 4, RTS is far down at the end of the teams window list, so you have to scroll down first, and also it's hard to find, because now it looks like any other window.

**Reply by adam-wood** [reply]
@ChadThomsen - If you are using Windows 11, there is a good chance if you hover the mouse cursor over RTS on the taskbar (While on a Teams call), you can simply select 'Share this window', Then 'Share'. See the gif below.

That being said, I would also appreciate an option for RTS to automatically start sharing so I don't need to press Play.

![2022 10 02 QuickShare](https://user-images.githubusercontent.com/1785141/193468742-ec49c67c-7ab8-4b09-a2ce-84a993beffe1.gif)

This [link](https://windowsreport.com/teams-screen-share-taskbar/) has more detail

**Reply by ChadThomsen** [reply]
@adam-wood - Good idea but I am running Windows 10 so that won't work.
@tom-englert - I am trying to eliminate a step, not re order them as that IMO does not make it any quicker or easier.

**Reply by tom-englert** [reply]
@adam-wood looks cool on Win11, does this also work properly when RTS is in active mode?

**Reply by tom-englert** [reply]
It's open source, anyone who thinks that this one click is worth the effort can contribute this feature.

**Reply by adam-wood** [reply]
> @adam-wood looks cool on Win11, does this also work properly when RTS is in active mode?

Sure does - I use it daily.
